src/app/components/Footer/index.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Paper, BottomNavigation, Fab } from '@mui/material';
+
+type Props = {
+  actionButtonIcon: JSX.Element;
+  onClickActionButton: () => void;
+};
+
+export default (props: Props): JSX.Element => {
```

Comment:
Why should the structure be that way? 
It's basically the DRY principle used in naming components since we already know that it is a component, and which component for the matter of fact.
I find it way more legible this way since I have an index.tsx in every folder. So when looking for the most important files in the folder, I don't have to look too closely. The other thing is if I would want to create another, smaller component inside this component (I'd do that if I, for example, wanted to use it only in this component, and nowhere else in the app, I would place it inside the same folder), I would know that the main file is index.tsx and the other files are smaller components used there.
Regarding the imports, I don't really see a problem with that either, mainly because it is short and everyone knows what it means.